sceneview/src/main/kotlin/io/github/sceneview/scene/TransformableManager.kt

```diff
@@ -0,0 +1,146 @@
+        listOf(dragGestureRecognizer, pinchGestureRecognizer, twistGestureRecognizer)
+
+    private val activeTransformationControllers =
+        mutableListOf<BaseTransformationController<*>>()
+
+    private var selectedNode: Node? = null
+
+    var translationControllerBuilder: TranslationControllerBuilder? = null
+
+    fun onNodeTap(node: Node?) {
+        selectedNode?.let { deselectNode() }
+        if (node !is Transformable || !node.isFocusable) return
+        activeTransformationControllers.forEach { it.cancel() }
+        activeTransformationControllers.clear()
+
+        node.editModes.forEach {
+            val gestureController = when (it) {
+                Transformable.EditMode.ROTATE -> RotationController(
+                    node,
+                    twistGestureRecognizer,
+                    this
+                )
+                Transformable.EditMode.SCALE -> ScaleController(node, pinchGestureRecognizer, this)
+                Transformable.EditMode.MOVE -> checkNotNull(
+                    translationControllerBuilder?.build(
+                        node,
+                        this,
+                        dragGestureRecognizer
+                    )
+                ) { "translationControllerBuilder needs to be set to use MOVE gesture." }
+            }
+            activeTransformationControllers.add(gestureController)
+        }
+        selectNode(node)
+    }
+
+    private fun selectNode(node: Node): Boolean {
+        if ((node as? Transformable)?.isFocusable != true) return false
+        if (!deselectNode()) {
+            return false
+        }
+        selectionVisualizer.applySelectionVisual(node)
+        selectedNode = node
+        return true
+    }
+
+    private fun deselectNode(): Boolean {
+        if (selectedNode == null) {
+            return true
+        }
+
+        if (selectedNodeIsTransforming()) {
+            return false
+        }
+
+        selectedNode?.let {
+            selectionVisualizer.removeSelectionVisual(it)
+        }
+        selectedNode = null
+
+        return true
+    }
+
+    private fun selectedNodeIsTransforming() =
+        activeTransformationControllers.any { it.isTransforming }
+
+    /** Dispatches touch events to the gesture recognizers contained by this transformation system.  */
+    fun onTouch(pickHitResult: PickHitResult?, motionEvent: MotionEvent?) {
+        if (selectedNode == null && pickHitResult?.node?.isFocusable == true) {
+            pickHitResult.node?.let { onNodeTap(it) }
+        } else {
+            gestureRecognizers.forEach {
+                it.onTouch(pickHitResult, motionEvent)
+            }
+        }
+    }
+
+    override fun isSelected(node: Node): Boolean = selectedNode === node
+
+    override fun select(node: Node): Boolean {
+        return (node as? Transformable)?.let { selectNode(node) } ?: false
+    }
+
+    interface TranslationControllerBuilder {
```

Comment:
Does a Builder is really needed. It introduce a lot of boiler plate code within Kotlin